csp_billing_adapter_microsoft/plugin.py

```diff
@@ -32,16 +32,12 @@
 
 log = logging.getLogger('CSPBillingAdapter')
 
-METADATA_ADDR = 'http://169.254.169.254/metadata'
-METADATA_API_VERSION = '?api-version=2021-12-13'
-METADATA_INSTANCE_PATH = '/instance'
-METADATA_TOKEN_PATH = '/identity/oauth2/token'
-METADATA_TOKEN_RESOURCE = '&resource=https://management.azure.com/'
-
-METADATA_INSTANCE_URL = METADATA_ADDR + METADATA_INSTANCE_PATH
-METADATA_TOKEN_URL = \
-    METADATA_ADDR + METADATA_TOKEN_PATH + METADATA_API_VERSION + \
-    METADATA_TOKEN_RESOURCE
+METADATA_URL = 'http://169.254.169.254/metadata/'
+COMPUTE_URL = METADATA_URL + 'instance/compute?api-version=2017-08-01'
+# version 2020-09-01 include license type
+# for more info check https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service?tabs=linux#attested-data
+SIGNATURE_URL = METADATA_URL + 'attested/document?api-version=2020-09-01'
+METADATA_HEADER = {'Metadata': 'True'}
```

Comment:
You removed the token variables I added but I think I will need those  per https://learn.microsoft.com/en-us/partner-center/marketplace/marketplace-metering-service-apis#metered-billing-single-usage-event when I generate a token.

Also is there a reason for this api-version, "2020-09-01" similar to my question above?